flow/serialize.h

```diff
@@ -170,6 +170,27 @@ inline void load( Archive& ar, std::set<T>& value ) {
 	ASSERT( ar.protocolVersion() != 0 );
 }
 
+template <class Archive, class K, class V>
+inline void save( Archive& ar, const std::map<K, V>& value ) {
+	ar << (int)value.size();
+	for (const auto &it : value) {
+		ar << it.first << it.second;
+	}
+	ASSERT( ar.protocolVersion() != 0 );
+}
+template <class Archive, class K, class V>
+inline void load( Archive& ar, std::map<K, V>& value ) {
+	int s;
+	ar >> s;
+	value.clear();
+	for (int i = 0; i < s; ++i) {
```

Comment:
value.reserve(s);

Reply:
value is a `std::map`, so reserve will not work here.

Reply:
makes sense